lib/ecto/schema.ex

```diff
@@ -1992,7 +1999,14 @@ defmodule Ecto.Schema do
     writable = opts[:writable] || :always
     put_struct_field(mod, name, Keyword.get(opts, :default))
 
-    if Keyword.get(opts, :redact, false) do
+    redact_field? = Keyword.get_lazy(opts, :redact, fn ->
+      case Module.get_attribute(mod, :schema_redact) do
+        :all_except_primary_keys -> not pk?
+        _ -> false
```

Comment:
So we raise with unknown values!

```suggestion
        nil -> false
```

Reply:
That causes the test for `@schema_redact false` to fail. 😅

Reply:
Or are you saying I should raise an exception here? And write a corresponding test? Can do!

Reply:
Indeed, I changed it to default to `false`. :) I think we are good now!